File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_user_store/lib/src/services/user_store_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/user_store_models.dart';

/// User Store service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class UserStoreService {
  const UserStoreService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'user-store';

  UserStorePageResult<Never> empty() =>
      const UserStorePageResult<Never>(items: <Never>[]);
}

File: apps/sdkwork-appstore-h5/packages/sdkwork-appstore-h5-user-store/src/services/userStoreService.ts
import type { AppstoreAppSdkClient } from '@sdkwork/appstore-h5-core';

import type { UserStorePageResult } from '../types/userStoreModels';

/**
 * User Store service.
 *
 * The generated app SDK clients are injected by the application root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export class UserStoreService {
  readonly capability = 'user-store';

  constructor(private readonly client: AppstoreAppSdkClient) {}

  empty(): UserStorePageResult<never> {
    return { items: [] };
  }
}
